### ISSUE TITLE
latest folding does not handle c-files properly

### ISSUE THREAD

**Issue by jdkloe** [jaalto/project-emacs--folding-mode#1]
I have a collection of c-source code files that use folding.el with these markers:
(folding-add-to-marks-list 'c-mode           "/\* #["  "/\* #]"  "*/")

This used to work just fine, but recently emacs crashes on these files. I am not sure if emacs or folding.el is to blame but would appreciate if you could take a look at the problem.
On the attached testcase I get the following behaviour:
-on startup: emacs does not fold the file, even though the mini-buffer says "Folding buffer... done" and no folding warnings or errors are mentioned in the _messages_ buffer.
-when manually folding the first region (I use [F1] for this) the region is not folded but garbled (see fold1.png). I can unfold (I use [F2] for this) and get my original unfolded view again.
-when manually folding the second region emacs just hangs and I have to kill it manually. It does not respond to any key or mouse click whatsoever. The second region itself is garbled after this action.
-when manually folding the third region the region is not folded but garbled (see fold3.png) and can be unfolded just as region 1.

I am experiencing this on fedora 19 and 20 for emacs versions:
emacs-24.3-11.fc19.x86_64  and emacs-24.3-13.fc20 
with folding-version-time "2014.0401.0703"

I invoke emacs like this:

```
   emacs --no-site-file -q -l dot_emacs -T 'folding bug' test_folding.c
```

I trimmed down my emacs settings file to a minimal example that shows the problem:

```
;; -*- folded-file: t; mode: Lisp; -*-

(message "* --[ Loading my personal Emacs init file ]--")

;; add some directories to loadpath
(setq load-path (cons  "./" load-path ))
;; keyboard customisation

;; go to major mode depending on file extension
(autoload 'c-mode       "cc-mode"       "C Editing Mode" t)
(autoload 'folding-mode "folding"       "minor mode: folding editor" t)

;; on startup goto major-mode depending on file-extension :
(setq auto-mode-alist
      (mapcar 'purecopy
          '(("\\.c$"     . c-mode)
        )
          )
      )

;; folding mode settings
(if (load "folding" 'nomessage 'noerror)
    (folding-mode-add-find-file-hook))

(folding-add-to-marks-list 'c-mode           "/* #["  "/* #]"  "*/")

;; set appropriate hook (only then folding is automatically activated)
(add-hook 'c-mode-hook                 'folding-mode )

;; function-key bindings :
(define-key key-translation-map [f1] nil) ;; Give me back my f1 key!
(global-set-key   [f1]        'folding-hide-current-entry )
(global-set-key   [f2]        'folding-show-current-entry )
(global-set-key   [f3]        'folding-whole-buffer )
(global-set-key   [f4]        'folding-open-buffer )

(custom-set-variables '(inhibit-startup-screen t))
```

![fold1](https://cloud.githubusercontent.com/assets/1906112/2679772/84092f1a-c178-11e3-985a-782cdd5a1e03.png)
![fold2](https://cloud.githubusercontent.com/assets/1906112/2679773/895a1a7e-c178-11e3-9571-0256a05606a7.png)
![fold3](https://cloud.githubusercontent.com/assets/1906112/2679774/8cdb931c-c178-11e3-91dc-a66caa644e4f.png)
![startup](https://cloud.githubusercontent.com/assets/1906112/2679775/90323d72-c178-11e3-96d4-f8bbf77925e4.png)

**Comment by nega0** [comment]
Sorry no one's reached out to you in the past 5 years... I hope you've since resolved your issue. 

Assuming you were using f738e28 (11 days old at the time of your post) and taking a minimal init
```lisp
(add-to-list 'load-path (expand-file-name "~/.emacs.d/lisp/folding-mode"))

(autoload 'folding-mode          "folding" "Folding mode" t)
(autoload 'turn-off-folding-mode "folding" "Folding mode" t)
(autoload 'turn-on-folding-mode  "folding" "Folding mode" t)

;; folding mode settings
(if (load "folding" 'nomessage 'noerror)
    (folding-mode-add-find-file-hook))

(folding-add-to-marks-list 'c-mode "/* #["  "/* #]"  "*/")
(add-hook 'c-mode-hook 'folding-mode )
```
and your example text

```C
/* -*- folded-file: t -*- */

/* #[ Description:
   this is a small collection of blahaahgbaslna
*/
/* #] */

/* #[ include files */
#include <stdio.h>

/* #] */

/* #[  do something */
int main(int i)
{ return 42; }

/* #] */
```
the buffer folds as expected.

```C
/* -*- folded-file: t -*- */

/* #[ Description:...

/* #[ include files */...

/* #[  do something */...
```
This is with both Emacs 22.1.1 and Emacs 25.3.50.1. Sorry.
Maybe your copy of `folding.el` was corrupt at the time (DOS line endings?)

**Comment by jdkloe** [comment]
thanks for your reply, and also sorry from my side for not reporting back. It seems I totally forgot I opened this issue al these years ago.
Yes, the issue has been resolved on my side so I will close this report.

Looking at my revision control history for my emacs modules I updated to the latest github version on the same day, and I guess this solved the issue for me. The folding version I installed that day has time code 2014.0401.0703 (copyright years up to 2013) and this works well until the present day. The older version with time code 2012.0514.2140 (with copyright years up to 2012) had the mentioned problem.
